### PR TITLE
[BugFix] Make query pool memory stats more accurate

### LINKED ISSUE #40965: query pool memory usage became negative
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
2. INSERT INTO '....'
3. SELECT  '....'

Create a iceberg table with many rows and many files. 

Run ` Select count(*) from iceberg.big_iceberg.lineitem_id_par`

```
CREATE EXTERNAL CATALOG `iceberg` PROPERTIES (
    "hive.metastore.uris"  =  "thrift://172.26.194.238:9083", 
    "type"  =  "iceberg", 
    "iceberg.catalog.type"  =  "hive" );

Select count(*) from iceberg.big_iceberg.lineitem_id_par

```

### Expected behavior (Required)

query pool memory usage is positive

### Real behavior (Required)

image

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by dirtysalt**
## Why I'm doing:

The root cause is:
- exec plan and scan morsels are allocated in `PInternalServiceImplBase<T>::_exec_plan_fragment`
- and these allocated objects are recorded into `process` mem tracker
- however those objects(especially scan morsels) are released in `ScanOperator::_close_chunk_source`
- and these allocated objects are recorded into `query_pool` mem tracker

So `query_pool` mem tracker will be negative. 

## What I'm doing:

switch to query context mem tracker when allocate exec plan and morsels.

**Known Problem**

However I found during a period time, query pool memory usage is larger than process memory usage. I'm not sure why. And this is a known problem. But when query ends, the query pool mem tracker goes zero. 

And that's because some objects are allocated and recorded into query pool mem tracker, are released and record into process mem tracker, so query pool mem tracker is higher than process. But I'm not sure which objects.

To resolve that problem, we have to trace every objects, and that needs huge effort. Oh dear god!

img_v3_027r_86b3c978-b21f-4e3d-8db4-4f3aea2b986g

Fixes https://github.com/StarRocks/starrocks/issues/40965

## What type of PR is this:

- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by imay — DISMISSED** [review]

File: be/src/exec/pipeline/fragment_executor.cpp [hunk 1]
```diff
@@ -709,11 +709,19 @@ Status FragmentExecutor::prepare(ExecEnv* exec_env, const TExecPlanFragmentParam
         SCOPED_RAW_TIMER(&profiler.prepare_runtime_state_time);
         RETURN_IF_ERROR(_prepare_workgroup(request));
         RETURN_IF_ERROR(_prepare_runtime_state(exec_env, request));
+
+        auto query_mem_tracker = _query_ctx->mem_tracker();
+        SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(query_mem_tracker.get());
```
Comment (satanson) [inline be/src/exec/pipeline/fragment_executor.cpp:714]: suggest to use FragmentContext.RuntimeState.instance_mem_tracker

File: be/src/exec/pipeline/fragment_executor.cpp [hunk 2]
```diff
@@ -709,11 +709,19 @@ Status FragmentExecutor::prepare(ExecEnv* exec_env, const TExecPlanFragmentParam
...
+
         RETURN_IF_ERROR(_prepare_exec_plan(exec_env, request));
         RETURN_IF_ERROR(_prepare_global_dict(request));
     }
     {
         SCOPED_RAW_TIMER(&profiler.prepare_pipeline_driver_time);
+
+        auto query_mem_tracker = _query_ctx->mem_tracker();
+        SCOPED_THREAD_LOCAL_MEM_TRACKER_SETTER(query_mem_tracker.get());
```
Comment (satanson) [inline be/src/exec/pipeline/fragment_executor.cpp:723]: as above

File: be/src/exec/pipeline/fragment_executor.cpp (hunk 1 above)
Comment (dirtysalt) [reply be/src/exec/pipeline/fragment_executor.cpp]: fixed.

File: be/src/exec/pipeline/fragment_executor.cpp (hunk 2 above)
Comment (dirtysalt) [reply be/src/exec/pipeline/fragment_executor.cpp]: fixed.

**Review by murphyatwork — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by dirtysalt** [comment]
@mergify backport branch-3.2

**Comment by dirtysalt** [comment]
@mergify backport branch-3.1

**Comment by dirtysalt** [comment]
@mergify backport branch-3.0

**Comment by dirtysalt** [comment]
@mergify backport branch-2.5

**Pull request merged**